backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofManager.java

```diff
@@ -136,6 +153,75 @@ public void sendPresentProofRequest(@NonNull UUID partnerId, @NonNull RequestPro
         }
     }
 
+    public void rejectPresentProofRequest(@NotNull PartnerProof proofEx, String explainString) {
+        if (PresentationExchangeState.REQUEST_RECEIVED.equals(proofEx.getState())) {
+            try {
+                sendPresentProofProblemReport(proofEx.getPresentationExchangeId(), explainString);
+                deletePartnerProof(proofEx.getId());
+            } catch (IOException e) {
+                throw new NetworkException("aca-py not available", e);
+            } catch (AriesException e) {
+                log.error("aca-py wallet item not found, attempting BPA delete");
+            }
+        } else {
+            throw new WrongApiUsageException("PresentationExchangeState != 'request-received'");
+        }
+    }
+
+    public void presentProof(@NotNull PartnerProof proofEx) {
+        if (PresentationExchangeRole.PROVER.equals(proofEx.getRole())
+                && PresentationExchangeState.REQUEST_RECEIVED.equals(proofEx.getState())) {
+            try {
+                ac.presentProofRecordsGetById(proofEx.getPresentationExchangeId())
+                        .ifPresent(record -> presentProof(record));
+                pProofRepo.updateState(proofEx.getId(), PresentationExchangeState.PRESENTATIONS_SENT);
+            } catch (IOException e) {
+                log.error("aca-py not reachable.", e);
+                return;
+            } catch (AriesException e) {
```

Comment:
Not needed as you throw it within a lambda, so it will never bubble up

Reply:
removed.